webapp/src/components/testing/Testing.vue

```diff
@@ -165,152 +185,186 @@
+const validationStore = useValidationStore();
+const memoryStore = useMemoryStore();
+const tasksStore = useTasksStore();
+const toaster = useToaster();
 
-const { step: stepRef, state: stateRef } = storeToRefs(validationStore)
-const currentTask = shallowRef<Task | undefined>(undefined)
+const { step: stepRef, state: stateRef } = storeToRefs(validationStore);
+const currentTask = shallowRef<Task>();
 
-const federatedTasks = computed<List<Task>>(() =>
-  tasksStore.tasks?.filter((t) => t.trainingInformation.scheme === 'federated').toList())
+const federatedTasks = computed(() =>
+  tasksStore.tasks
+    .filter((t) => t.trainingInformation.scheme === "federated")
+    .toList(),
+);
+const labels = computed(() =>
+  Set(currentTask.value?.trainingInformation.LABEL_LIST),
+);
+const memory = computed(() =>
+  memoryStore.useIndexedDB ? new IndexedDB() : new EmptyMemory(),
+);
 
-const memory = computed<Memory>(() =>
-  memoryStore.useIndexedDB ? new IndexedDB() : new EmptyMemory())
+const imageDataset = ref<NamedLabeledImageDataset>();
+const tabularDataset = ref<Dataset<Tabular>>();
+const textDataset = ref<Dataset<Text>>();
+const dataset = computed<TypedNamedLabeledDataset | undefined>(() => {
+  if (
+    Set.of<unknown>(
+      imageDataset.value,
+      tabularDataset.value,
+      textDataset.value,
+    ).filter((d) => d !== undefined).size > 1
+  )
+    throw new Error("multiple dataset entered");
 
-const datasetBuilder = computed<data.DatasetBuilder<File> | undefined>(() => {
-  if (currentTask.value === undefined) {
-    return undefined
-  }
-  let dataLoader: data.DataLoader<File>
-  switch (currentTask.value.trainingInformation.dataType) {
-    case 'tabular':
-      dataLoader = new WebTabularLoader(currentTask.value, ',')
-      break
-    case 'image':
-      dataLoader = new WebImageLoader(currentTask.value)
-      break
-    case 'text':
-      dataLoader = new WebTextLoader(currentTask.value)
-      break
-    default:
-      throw new Error(`Browser data loader for data type ${currentTask.value.trainingInformation.dataType} is not implemented`)
+  if (imageDataset.value !== undefined) return ["image", imageDataset.value];
+  if (tabularDataset.value !== undefined)
+    return ["tabular", tabularDataset.value];
+  if (textDataset.value !== undefined) return ["text", textDataset.value];
+
+  return undefined;
+});
+
+watch(tabularDataset, async (dataset) => {
+  if (dataset === undefined) return;
+  if (currentTask.value === undefined)
+    throw new Error("dataset entered but no task");
+
+  const wantedColumns = Set(currentTask.value.trainingInformation.inputColumns);
+
+  try {
+    for await (const [columns, i] of dataset
+      .map((row) => Set(Object.keys(row)))
+      .zip(Range(1)))
+      if (!columns.isSuperset(wantedColumns))
+        throw new Error(
+          `row ${i} is missing columns ${wantedColumns.subtract(columns).join(", ")}`,
+        );
+  } catch (e) {
+    let msg = "Error when loading CSV";
```

Comment:
When connecting the same titanic csv I used during training at the testing step, I get the error: `Error when loading CSV: can't access private field or method: object is not the right class`

Reply:
Same error if I connect a misformatted csv

Reply:
and same fix as before, was a bit fast on this one :sweat_smile: